pystac/cache.py

```diff
@@ -59,6 +59,18 @@ class ResolvedObjectCache:
             to collections.
     """
 
+    id_keys_to_objects: Dict[str, "STACObject_Type"]
+    """Existing cache of a key made up of the STACObject and it's parents IDs mapped
```

Comment:
Should we [use `typing.Annotated` here](https://stackoverflow.com/a/8820636/96588)?

Reply:
#687 adds the classes from the `cache` submodule to the API docs, so when this PR has merged with those changes these attributes will already appear in the docs. Since it seems like the main purpose of `typing.Annotated` is to attach documentation to variables it doesn't seem like we would gain much by using it here and it would make the type annotations more complicated. Would there be other advantages to using `typing.Annotated` that I'm missing?

Reply:
Sounds good; I expected `typing.Annotated`'s main purpose was exactly that. I've not used it myself, I'm basically just a fan of explicitly tying the code and documentation together rather than using comments :)